src/App.jsx

```diff
@@ -14,33 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SortBy = {
+  sortByAlphabet: 'alphabet',
+  sortByLength: 'length',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPrepareGoods = (goods, { sortField, reversed }) => {
+  const prepareGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SortBy.sortByAlphabet:
+        prepareGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SortBy.sortByLength:
+        prepareGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        return 0;
+    }
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPrepareGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const handleReset = () => {
+    setSortField('');
+    setReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortBy.sortByAlphabet,
+          })}
+          onClick={() => setSortField(SortBy.sortByAlphabet)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortBy.sortByLength,
+          })}
+          onClick={() => setSortField(SortBy.sortByLength)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
```

Comment:
See how to update the state based on the current state value properly:
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state